core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/repository/BaseRepositoryPreCommitHook.java

```diff
@@ -42,12 +43,22 @@ public BaseRepositoryPreCommitHook(Logger log) {
 	
 	@Override
 	public void run(StagingArea staging) {
+		boolean needsDocumentUpdate;
+		if (staging.getContext() instanceof RepositoryTransactionContext) {
+			RepositoryTransactionContext transactionContext = (RepositoryTransactionContext) staging.getContext();
+			needsDocumentUpdate = !DatastoreLockContextDescriptions.CREATE_VERSION.equals(transactionContext.parentLock());
+		} else {
+			needsDocumentUpdate = true;
+		}
+
 		staging.read(index -> {
-			updateDocuments(staging, index);
+			if (needsDocumentUpdate) {
```

Comment:
It would be better to prevent the entire read transaction, just in case. Also, is there a subclass implementation that overrides this method? Or a subclass implementation where the updateDocuments is required to be called upon versioning?